x-pack/test/rule_registry/security_and_spaces/tests/basic/update_alert.ts

```diff
@@ -0,0 +1,266 @@
+        .post(`${getSpaceUrlPrefix()}${TEST_URL}`)
+        .auth(superUser.username, superUser.password)
+        .set('kbn-xsrf', 'true')
+        .send({
+          ids: ['this id does not exist'],
+          status: 'closed',
+          index: APM_ALERT_INDEX,
+          _version: Buffer.from(JSON.stringify([0, 1]), 'utf8').toString('base64'),
+        })
+        .expect(404);
+    });
+
+    it('should return a 404 when superuser accesses not-existent alerts as data index', async () => {
+      await supertestWithoutAuth
+        .post(`${getSpaceUrlPrefix()}${TEST_URL}`)
+        .auth(superUser.username, superUser.password)
+        .set('kbn-xsrf', 'true')
+        .send({
+          ids: [APM_ALERT_ID],
+          status: 'closed',
+          index: 'this index does not exist',
+          _version: Buffer.from(JSON.stringify([0, 1]), 'utf8').toString('base64'),
+        })
+        .expect(404);
+    });
+
+    function addTests({ space, authorizedUsers, unauthorizedUsers, alertId, index }: TestCase) {
+      authorizedUsers.forEach(({ username, password }) => {
+        it(`${username} should be able to update alert ${alertId} in ${space}/${index}`, async () => {
+          await esArchiver.load('x-pack/test/functional/es_archives/rule_registry/alerts'); // since this is a success case, reload the test data immediately beforehand
+          await supertestWithoutAuth
+            .post(`${getSpaceUrlPrefix(space)}${TEST_URL}`)
+            .auth(username, password)
+            .set('kbn-xsrf', 'true')
+            .send({
+              ids: [alertId],
+              status: 'closed',
+              index,
+              _version: ALERT_VERSION,
+            })
+            .expect(200);
+        });
+
+        it(`${username} should fail to update alert ${alertId} in ${space}/${index} with an incorrect version`, async () => {
+          await supertestWithoutAuth
+            .post(`${getSpaceUrlPrefix(space)}${TEST_URL}`)
+            .auth(username, password)
+            .set('kbn-xsrf', 'true')
+            .send({
+              ids: [alertId],
+              status: 'closed',
+              index,
+              _version: Buffer.from(JSON.stringify([999, 999]), 'utf8').toString('base64'),
+            })
+            .expect(409);
+        });
+
+        // // NOTE: this test case actually results in a 500 error, not 404
+        // it(`${username} should fail to update a non-existent alert in ${space}/${index}`, async () => {
+        //   const fakeAlertId = 'some-alert-id-that-doesnt-exist';
+        //   await supertestWithoutAuth
+        //     .post(`${getSpaceUrlPrefix(space)}${TEST_URL}`)
+        //     .auth(username, password)
+        //     .set('kbn-xsrf', 'true')
+        //     .send({
+        //       ids: [fakeAlertId],
+        //       status: 'closed',
+        //       index,
+        //       _version: ALERT_VERSION,
+        //     })
+        //     .expect(404);
+        // });
```

Comment:
If you remove the superuser-specific tests above, you can uncomment this and add an additional case for missing index that will be exercised for all users:
```suggestion
        it(`${username} should fail to update a non-existent alert in ${space}/${index}`, async () => {
          const fakeAlertId = 'some-alert-id-that-doesnt-exist';
          await supertestWithoutAuth
            .post(`${getSpaceUrlPrefix(space)}${TEST_URL}`)
            .auth(username, password)
            .set('kbn-xsrf', 'true')
            .send({
              ids: [fakeAlertId],
              status: 'closed',
              index,
              _version: ALERT_VERSION,
            })
            .expect(404);
        });

        it(`${username} should fail to update alert ${alertId} in a non-existent index in ${space}`, async () => {
          const fakeIndexId = 'some-index-that-doesnt-exist';
          await supertestWithoutAuth
            .post(`${getSpaceUrlPrefix(space)}${TEST_URL}`)
            .auth(username, password)
            .set('kbn-xsrf', 'true')
            .send({
              ids: [alertId],
              status: 'closed',
              index: fakeIndexId,
              _version: ALERT_VERSION,
            })
            .expect(404);
        });
```